### ISSUE TITLE
Checkpoints should work for defenders in gauntlet-race

### ISSUE THREAD

**Issue by JeffBobbo** [red-eclipse/base#153]
Currently in gauntlet-race checkpoints don't do anything for the defender.

This should be changed so that they function as normal, with exception to the last one not rewarding any points/time for the completion of the lap.

Simply because:
* Dying as a defender causes you to respawn at the start of the map, meaning you can't catch up
* In non-endurance (default), attackers get impulse refill which defenders don't, meaning they can do the lap faster than you

**Comment by bonifarz** [comment]
Yeah, I remember this discussion in-game. This suggestion includes the teleport functionality of the last checkpoint, right?

Another thought, which may be a bit more complicated to implement: Whenever a checkpoint is passed by a runner, it serves as spawn point for the defending team. This might work even better, especially with small teams.

**Comment by JeffBobbo** [comment]
@bonifarz I don't think that's such a good idea, as it means the defenders are essentially defending against the lead, where as an organized team could stage their members at each checkpoint to slow down people as they come through (rather than trying to constantly play catch up).

**Comment by bonifarz** [comment]
> defending against the lead

No, that's not what I wanted to suggest.
> latest checkpoint passed by a runner

So, any following runner counts, not only the "lead", and  defenders spawn on the most recent checkpoint triggered. I thought this would be much fairer than "randomized" spawning, as with several player starts.
However, your comment makes me realize that runners could abuse these mechanics: If they keep suiciding after an early checkpoint, they can keep the defenders from spawning on the later checkpoints.

Your suggestion is fine, I just wanted to add something to discuss.

**Comment by bonifarz** [comment]
http://redeclipse.net/forum/viewtopic.php?f=4&t=678
> ... it's a bug. Omega guys are supposed to spawn at random checkpoints.
